.pre-commit-config.yaml

```diff
@@ -0,0 +1,60 @@
+default_language_version:
+  python: python3.7
+
+fail_fast: true
+
+repos:
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.2.0
+  hooks:
+    - id: check-ast   # checks Python syntax
+    - id: check-json  # checks JSON syntax
+    - id: check-yaml  # checks YAML syntax
+    - id: check-toml  # checks TOML syntax
+    - id: end-of-file-fixer  # checks there is a newline at the end of the file
+    - id: trailing-whitespace  # trims trailing whitespace
+    - id: check-merge-conflict  # checks for no merge conflict strings
+    - id: check-shebang-scripts-are-executable  # checks all shell scripts have executable permissions
+    - id: mixed-line-ending  # normalizes line endings
+    - id: no-commit-to-branch  # prevents committing to master
+    - id: pretty-format-json  # indents and sorts JSON files
+
+- repo: https://github.com/psf/black
+  rev: 22.6.0  # keep this aligned with the black version in setup.cfg
+  hooks:
+  - id: black
+
+# These can fail if some dependencies are missing. Also untested on Windows.
+- repo: local
```

Comment:
note: I avoided the `local` one repo types as some people (e.g. @ArzelaAscoIi ) use dev machines/containers which means they might not have the dependencies installed in the environment where they commit.

Reply:
Yes I have considered this. However I think we can't avoid it in our situation. I'd rather ask these people not to install the hook in environments where the deps can't be installed.

I also considered having two hooks that can be installed separately, but haven't looked into that yet. I think this can already be tried out for us as it is, if it causes too much trouble of course we'll revisit 🙂

Reply:
makes sense 👍🏻